src/main/java/com/nimbusds/jose/crypto/impl/ECDH1PU.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * nimbus-jose-jwt
+ *
+ * Copyright 2012-2021, Connect2id Ltd and contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.nimbusds.jose.crypto.impl;
+
+
+import com.nimbusds.jose.*;
+import com.nimbusds.jose.crypto.utils.ECChecks;
+import com.nimbusds.jose.jwk.Curve;
+import com.nimbusds.jose.jwk.OctetKeyPair;
+import com.nimbusds.jose.util.Base64URL;
+import com.nimbusds.jose.util.ByteUtils;
+
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.charset.StandardCharsets;
+import java.security.interfaces.ECPrivateKey;
+import java.security.interfaces.ECPublicKey;
+
+
+/**
+ * Elliptic Curve Diffie-Hellman One-Pass Unified Model (ECDH-1PU)
+ * key agreement functions and utilities.
+ *
+ * @see <a href="https://datatracker.ietf.org/doc/html/draft-madden-jose-ecdh-1pu-04">Public Key Authenticated Encryption for JOSE: ECDH-1PU</a>
+ *
+ * @author Alexander Martynov
+ * @version 2021-08-03
+ */
+public class ECDH1PU {
+
+    /**
+     * Resolves the ECDH algorithm mode.
+     *
+     * @param alg The JWE algorithm. Must be supported and not
+     *            {@code null}.
+     *
+     * @return The algorithm mode.
+     *
+     * @throws JOSEException If the JWE algorithm is not supported.
+     */
+    public static ECDH.AlgorithmMode resolveAlgorithmMode(final JWEAlgorithm alg)
+        throws JOSEException {
+
+        if (alg.equals(JWEAlgorithm.ECDH_1PU)) {
+
+            return ECDH.AlgorithmMode.DIRECT;
+
+        } else if (alg.equals(JWEAlgorithm.ECDH_1PU_A128KW) ||
```

Comment:
`else if` can be reduced to `if` and `else` removed as they're all return/throw branches